### PR TITLE
chore: Set permissions for GitHub actions

### PR DESCRIPTION
**Pull request by naveensrinivasan**
Restrict the GitHub token permissions only to the required ones; this way, even if the attackers will succeed in compromising your workflow, they won’t be able to do much.

- Included permissions for the action. https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

[Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/)

Signed-off-by: naveen <172697+naveensrinivasan@users.noreply.github.com>

### CODE REVIEW

File: .github/workflows/issue-comment-created.yml [hunk 1]
```diff
@@ -7,8 +7,13 @@ on:
 env:
   IN_MAINTAINER_LIST: ${{ contains(secrets.MAINTAINER_LIST, github.actor) }}
 
+permissions:
+  contents: read
+
 jobs:
   issue_comment_triage:
+    permissions:
+      contents: none
```
Comment (justinretzolk) [inline .github/workflows/issue-comment-created.yml:16]: In order to remove labels, we need write permissions

```suggestion
    permissions:
      issues: write
```

File: .github/workflows/lock.yml [hunk 2]
```diff
@@ -4,8 +4,14 @@ on:
   schedule:
     - cron: '50 1 * * *'
 
+permissions:
+  contents: read
+
```
Comment (justinretzolk) [inline .github/workflows/lock.yml:9]: I believe we can omit this entirely given the permissions declaration in the `job.lock.permissions` sections.

```suggestion
```

File: .github/workflows/issues.yml [hunk 3]
```diff
@@ -7,6 +7,9 @@ on:
 env:
   IN_MAINTAINER_LIST: ${{ contains(secrets.MAINTAINER_LIST, github.actor) }}
 
+permissions:
+  contents: read
+
```
Comment (justinretzolk) [inline .github/workflows/issues.yml:12]: This workflow uses a custom token, and so we should disable all scopes.

```suggestion
permissions: {}
```

File: .github/workflows/maintainer-edit.yml [hunk 4]
```diff
@@ -3,8 +3,14 @@ on:
   pull_request_target:
     types:
       - opened
+permissions:
+  contents: read
+
```
Comment (justinretzolk) [inline .github/workflows/maintainer-edit.yml:8]: I don't believe we need content read permissions for this workflow, and permissions for the job are defined below, so we should be able to omit this.

```suggestion
```

File: .github/workflows/pull_requests.yml [hunk 5]
```diff
@@ -8,6 +8,9 @@ env:
 
 jobs:
   Labeler:
+    permissions:
+      contents: read  # for actions/labeler to determine modified files
+      pull-requests: write  # for actions/labeler to add labels to PRs
```
Comment (justinretzolk) [inline .github/workflows/pull_requests.yml:13]: The `actions/labeler` action is using a custom secret in this case, so we don't need to give it pull request write permissions.

```suggestion
```

File: .github/workflows/pull_requests.yml [hunk 6]
```diff
@@ -17,6 +20,9 @@ jobs:
           configuration-path: .github/labeler-pr-triage.yml
           repo-token: ${{ secrets.GITHUB_TOKEN }}
   NeedsTriageLabeler:
+    permissions:
+      contents: read  # for actions/labeler to determine modified files
+      pull-requests: write  # for actions/labeler to add labels to PRs
```
Comment (justinretzolk) [inline .github/workflows/pull_requests.yml:25]: The `actions/labeler` action is using a custom secret in this case, so we don't need to give it pull request write permissions.

```suggestion
```

File: .github/workflows/pull_requests.yml [hunk 7]
```diff
@@ -27,6 +33,8 @@ jobs:
           configuration-path: .github/labeler-pr-needs-triage.yml
           repo-token: ${{ secrets.GITHUB_TOKEN }}
   SizeLabeler:
+    permissions:
+      pull-requests: write  # for bflad/pr-size-labeler to add labels & comment on PRs
```
Comment (justinretzolk) [inline .github/workflows/pull_requests.yml:37]: The `bflad/pr-size-labeler` action is using a custom secret in this case, so we don't need to give it pull request write permissions.

```suggestion
```

File: .github/workflows/stale.yml [hunk 8]
```diff
@@ -3,8 +3,14 @@ on:
...
     - cron: "40 17 * * *"
 
+permissions:
+  contents: read
+
 jobs:
   stale:
+    permissions:
+      issues: write  # for actions/stale to close stale issues
+      pull-requests: write  # for actions/stale to close stale PRs
```
Comment (justinretzolk) [inline .github/workflows/stale.yml:13]: The `actions/stale` action is using a custom secret, so we don't need to grant additional permissions in this case.

```suggestion
```

File: .github/workflows/stale.yml [hunk 9]
```diff
@@ -3,8 +3,14 @@ on:
   schedule:
     - cron: "40 17 * * *"
 
+permissions:
+  contents: read
+
```
Comment (justinretzolk) [inline .github/workflows/stale.yml:8]: Given that the only job in this workflow is using a custom token, we should limit all scopes.

```suggestion
permissions: {}

```

**Review by justinretzolk — CHANGES_REQUESTED** [review]
Thank you very much for taking the time to raise this! This is definitely something I'd like to work on getting merged, but I left a few comments I'd like to see addressed before we do so.

**Comment by naveensrinivasan** [comment]
@justinretzolk   I don't have time to work on this PR now. I would request you, please, take it over. Thanks

**Comment by justinretzolk** [comment]
Hey @naveensrinivasan, no problem at all! Thank you for letting me know. We'll gladly take over from here.

**Pull request closed without merging**